### DISCUSSION TITLE
Failed to connect to server

### DISCUSSION THREAD

**Discussion by asimic94** [KL-Psychological-Methodology/ESMira#11] [Questions] [answered]
First of all, thank you for working on and developing an amazing open-source application for running experience sampling studies. This app will surely help hundreds of ESM researchers (including me).

I have decided to use your application to design a forthcoming ESM study. However, after following your instruction on GitHub, I ran into some unexpected problems when starting the interface screen (see attached screenshot). Unfortunately, I am a complete newbie here, and it is obvious I made a mistake in the earlier steps. I would really appreciate any kind of help here!

Thank you in advance,

Andrej Simić

![image](https://github.com/KL-Psychological-Methodology/ESMira/assets/161062749/9658d768-6d09-485e-9b92-3fa33b0ef115)

**Comment by JodliDev** [comment]
Hello

I hope you can put ESMira to good use :)

Judging from the slightly different error in the email you sent me, I assume that there is some kind of server error that ESMira is unable to understand. I assume PHP is showing some errors.
Would it be possible to send me the link to the page where this error occurs? Ill need to see the complete server resonse to understand what is happening.

**Reply by asimic94** [reply]
Hi! Thanks for the response and the help. You can access the page here: https://psychuntz.ba/ESMira-web/

Would you need anything else?

**Reply by JodliDev** [reply]
Your server seems to not have mod_rewrite enabled. And the script that usually checks for that, seems to produce an unexpected error.
There should be an easy fix for that.
I will be able to give you a more detailed response tomorrow :)

**Reply by asimic94** [reply]
Thanks! You really helped me a lot. I've been trying to find a fix for about a week.

**Reply by JodliDev** [reply]
Ok. Sorry for the delay. I had to make sure I am not telling you things that are wrong :) .
The fact that the site is failing with an error is our fault. When checking your server configuration, it tried to give you feedback on missing features, but that causes a bug (and since these features are usually enabled, we never noticed that bug until now).

To help you: Your server is missing two features: Htaccess and mod_rewrite (technically both of them are optional and if it werent for the bug, you should have been able to set up ESMira anyway. Though I recommend enabling them for security and convenience reasons).
We have a section in our Wiki on how you can enable them. You can find that section here (if you scroll up you will also find a short explaination why these features are necessary):
https://github.com/KL-Psychological-Methodology/ESMira/wiki/Setting-up-a-server#apache-server

Tell me if the Wiki has enough information for you to solve the problem. If not, I can give you more pointers :)

Ps
The bug you encountered will be fixed in the next version of ESMira.

**Reply by asimic94** [reply]
Thanks for the additional clarifications. Unfortunately, I am using a web hosting service that allows me to edit my data only via "cPanel." Therefore, I am not able to modify my server via the console. Is there a possible workaround to this issue? Thank you once again for your help!

**Reply by JodliDev** [reply]
I have never used cPanel. But I assume that these instructions are correct:

For htaccess (also called AllowOverideAll in settings):
https://bobcares.com/blog/allowoverride-all-cpanel/

For mod_rewrite (I assume that this is also disabled):
https://www.keliweb.com/billing/knowledgebase/8555/How-to-set-the-modrewrite-using-cPanel.html

**Reply by asimic94** [reply]
Unfortunately, now I'm getting a 500 Internal Server Error with the following message:  /home/psychunt/public_html/ESMira-web/.htaccess: AllowOverride not allowed here

**Reply by asimic94** [reply]
Hi,

I'm sorry for bothering you once again. I installed ESMira manually via a local server connected to the internet. I checked everything else, and it works perfectly. Therefore, ESMira might not run on servers that do not allow the AllowOverride command. 

So, thanks for all the help, and I will gladly give you feedback during the study design/data collection phases.

**Reply by JodliDev** [reply, marked as answer]
I am glad you got to test it successfully :)
In case you want to fix your public server as well:

I am quite sure that the new error message you are receiving is not caused by ESMira but stems from a misconfiguration of your server.
Looking at the instructions I have sent you, I am unsure if they are correct. Did they make you change `/home/psychunt/public_html/ESMira-web/.htaccess`? If yes then no wonder you have an error message now. Unless cPanel changed how Apache works, that would be wrong.
You need to change the Apache server settings for your public folder (there should be a setting for that somewhere in cPanel. This is a very common usecase).
There seems to be a cPanel Module that you can install:
https://docs.cpanel.net/ea4/apache/apache-module-cpanel/

As far as I can find out about cPanel, AllowOverride should be enabled by default. Maybe you need to install ESMira in `home/psychunt/public_html` and not in `home/psychunt/public_html/ESMira-web` ? (even though I am unsure what that changes).
Can you check if there is already an `.htaccess` file in your public folder (so ` /home/psychunt/public_html/.htaccess`)? Maybe it is overriding your settings.

(again, I can only guess since I have no experience with cPanel. But AllowOverride is a very common setting and should not be that difficult to enable - actually I am surprised that it is not enabled for you by default)
